File: oneAPI_workshop/day2/1300-1430-LAB3-hands-on-openmp/openmp-fortran/00-Hello/00-Hello.f90
program Hello
  use omp_lib

  ! make the print part parallel
  ! use !$omp parallel
  ! and set the number of threads with the call
  ! omp_set_num_threads()
  ! play a bit around with the parameter NT
  ! can you see concurrency?

  integer, parameter :: NT = 4

  call omp_set_num_threads(NT)

  !$omp parallel
    print*, "Hello "
    print*, "World "
  !$omp end parallel

end program Hello

File: oneAPI_workshop/day2/1300-1430-LAB3-hands-on-openmp/openmp-cpp/00-Hello/00-Hello.cpp
#include<iostream>
#include<omp.h>

#define MAX_NUM_THREADS 16

// In this program we have a parallel block
// which is indicated by the pragma statement.
// We have set the number of threads to 16
// but the instructions executed by each thread
// are unordered, this is the meaning of concurrency.

int main()
{
    omp_set_num_threads(MAX_NUM_THREADS);
	#pragma omp parallel
	{
		std::cout << "HELLO ";
		std::cout << "SCHLUMBERGER \n";

	}

	return 0;
}

// Concurrency defines operations that can execute in any order.
// Parallelism brings multiple hardware elements into the
// conversation so operations run at the same time.

